### DISCUSSION TITLE
Full-page components called "Directory" throw ComponentNotFoundException on refresh.

### DISCUSSION THREAD

**Discussion by syntheticminds** [livewire/livewire#8248] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.9

### Laravel version

v11.0.7

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Run `php artisan livewire:make Directory` to create the component.
2. Register it in routes/web.php `Route::get('/directory', Directory::class)->name('directory');`
3. Create a refresh button in the template.
```
<div>
    <button type="button" wire:click="$refresh">Click me</button>
</div>
```
4. Go to /directory
5. Click the button and see the error message: "Unable to find component: [directory]"

### Repository or Wirebox URL

https://wirebox.app/b/yrn05

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PerryvanderMeer** [comment, marked as answer]
See https://github.com/livewire/livewire/pull/8253

**Reply by syntheticminds** [reply]
Great work. Thanks!
